mqtt/apps/MQTT Data Monitor/appy.py

```diff
@@ -0,0 +1,1521 @@
+        if result['success']:
+            return jsonify(result), 200
+        else:
+            return jsonify(result), 400
+        
+    except Exception as e:
+        return jsonify({
+            'success': False,
+            'message': 'Connection test failed',
+            'error': str(e)
+        }), 500
+
+
+@flask_app.route('/api/start-mqtt-job', methods=['POST'])
+def start_mqtt_job():
+    """API endpoint to start the MQTT data ingestion job with user configuration"""
+    try:
+        # Get the configuration from the request
+        mqtt_config = request.json
+        
+        # Validate required fields
+        if not mqtt_config.get('broker_address'):
+            return jsonify({
+                'success': False,
+                'error': 'Broker address is required'
+            }), 400
+        
+        if not mqtt_config.get('catalog') or not mqtt_config.get('schema') or not mqtt_config.get('table'):
+            return jsonify({
+                'success': False,
+                'error': 'Catalog, Schema, and Table name are required'
+            }), 400
+        
+        # Get notebook_path and cluster_id from config
+        notebook_path = mqtt_config.get('notebook_path', '/Workspace/Users/jeffery.annor@databricks.com/mqtt/MQTT_data_source_v0')
+        cluster_id = mqtt_config.get('cluster_id', '0709-132523-cnhxf2p6')
+        
+        # Create the job
+        created_job = create_job(client, notebook_path, cluster_id)
+        
+        # Run the job with user configuration
+        run = run_job(client, created_job, mqtt_config)
+        
+        catalog = mqtt_config.get('catalog')
+        schema = mqtt_config.get('schema')
+        table = mqtt_config.get('table')
+        
+        return jsonify({
+            'success': True,
+            'job_id': created_job.job_id,
+            'run_id': run.run_id,
+            'message': f'MQTT data ingestion job started successfully. Data will be written to {catalog}.{schema}.{table}'
+        })
+        
+    except Exception as e:
+        return jsonify({
+            'success': False,
+            'error': str(e)
+        }), 500
+
+
+@flask_app.route('/api/refresh-data', methods=['POST'])
+def refresh_data():
+    """API endpoint to refresh dashboard data from specified table"""
+    global curr_data
+    try:
+        data = request.json
+        catalog = data.get('catalog')
+        schema = data.get('schema')
+        table = data.get('table')
+        # warehouse_id = data.get('warehouse_id', DEFAULT_WAREHOUSE_ID)
+        
+        # Validate required fields
+        if not catalog or not schema or not table:
+            return jsonify({
+                'success': False,
+                'error': 'Catalog, Schema, and Table name are required'
+            }), 400
+        
+        # Build the query
+        query = f"SELECT message, is_duplicate, qos, topic, received_time FROM {catalog}.{schema}.{table} ORDER BY received_time DESC LIMIT 100"
```

Comment:
F strings based dynamic sql generation pose a risk of sql injection attack. Try parameterized queries.